### PR TITLE
feat: add user-context

### PR DESCRIPTION
**Pull request by AdrianValdes**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     ],
     "rules": {
         "import/prefer-default-export": "off",
-        "import/no-default-export": "error"
+        "import/no-default-export": "error",
+        "react/jsx-props-no-spreading": "off"
```
Comment (chmette) [inline .eslintrc.json:31]: wow interesting, that they have this rule in there

File: src/index.jsx [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { UserProvider } from './context/userContext';
 import { App } from './App';
 import { reportWebVitals } from './reportWebVitals';
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <UserProvider>
```
Comment (chmette) [inline src/index.jsx:9]: why are you going with strict mode?

File: src/hooks/useInput.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { useState } from 'react';
+
+export const useInput = (initialValue) => {
+  const [value, setValue] = useState(initialValue);
+  return [
```
Comment (chmette) [inline src/hooks/useInput.js:5]: could you be so kind and just explain these hook to me? i understand what it does, but not step by step

**Review by chmette — APPROVED** [review]
few questions but approved.

File: src/index.jsx (hunk 2 above)
Comment (AdrianValdes) [reply src/index.jsx:9]: https://stackoverflow.com/questions/53183362/what-is-strictmode-in-react

Comment (AdrianValdes) [reply src/index.jsx:9]: https://reactjs.org/docs/strict-mode.html#:~:text=StrictMode%20is%20a%20tool%20for,and%20warnings%20for%20its%20descendants.

File: src/hooks/useInput.js (hunk 3 above)
Comment (AdrianValdes) [reply src/hooks/useInput.js:5]: we can talk about it later, it is faster.

Comment (chmette) [reply src/hooks/useInput.js:5]: allright

File: src/index.jsx (hunk 2 above)
Comment (chmette) [reply src/index.jsx:9]: ah ok, thanks i did not know these

**Pull request merged**